admin/class-admin-init.php

```diff
@@ -509,6 +484,51 @@ public function permalink_settings_notice() {
 		}
 	}
 
+	/**
+	 * Determines whether and where the "blog not public" notice should be displayed.
+	 *
+	 * @return bool Whether the "blog not public" notice should be displayed.
+	 */
+	private function should_display_discourage_search_engines_enabled_notice() {
+		return (
+			! $this->is_site_public()
+			&& WPSEO_Capability_Utils::current_user_can( 'manage_options' )
+			&& WPSEO_Options::get( 'ignore_blog_public_notification', false ) === false
+			&& (
+				$this->on_wpseo_admin_page()
+				|| in_array( $this->pagenow, [
+					'index.php',
+					'plugins.php',
+					'update-core.php',
+				], true )
+			)
+		);
+	}
+
+	/**
+	 * Displays an admin notice when WordPress is set to discourage search engines from indexing the site.
+	 *
+	 * @return void
+	 */
+	public function discourage_search_engines_enabled_notice() {
```

Comment:
Consider renaming these functions to `should_display_blog_public_notice()` and `blog_public_notice`, since also `ignore_blog_public_notification` and `'isBlogPublicNoticeDismissed'` are used. Also possible to rename the latter to something containing `discourage_search_engines_enabled`, but I think the shorter option is clearer. If you think `blog_public_notice` is not a good name since it regards the site, another option would be to use something like `blocking_robots_notice` or `site_not_public_notice`. What do you think?

Reply:
Yeah, naming is hard. In a first moment I was going to name those based on `blog_public` but that doesn't really clarify what they do. `blog_public` is the name of the WordPress option in the database. It's.. "ancient" :D and still refers to a "blog". Regardless, our notification warns users when the site it NOT public so at the very least our naing should be based on `blog_not_public`. Eventually, I thought that even `blog_not_public` wasn't clear enough. Instead, everyone knows what "discourage search engines" is in the WordPress admin so I'd be inclined to rename based on that.

Reply:
Do you then agree to rename `'isBlogPublicNoticeDismissed'` etc to `isDiscourageSearchEnginesEnabledDismissed`? To make it more clear it's about the same notification? Or maybe exclude the "enabled" part, to make it a bit shorter?

Reply:
Yes, I meant to rename as much as we can based on "discourage search engines", which seems the clearest option to me 🙂